packages/truffle-db/src/workspace/test/index.ts

```diff
@@ -249,274 +276,221 @@ mutation AddCompilation($compilerName: String!, $compilerVersion: String!, $sour
+          contents
+        }
+        ast {
+          json
         }
-      }
-      sources {
-        sourcePath
-        contents
-        ast  
       }
     }
-  }
 }`
 
-const AddContractConstructors = `
-mutation addConstructors($bytecodeId: ID!, $compilationId: ID!, $contractId: ID!) {
-    contractConstructorsAdd(input: {
-      contractConstructors:[{
+const AddContracts = gql`
+mutation addContracts($contractName: String, $compilationId: ID!, $bytecodeId:ID!, $abi:String!) {
+  contractsAdd(input: {
+    contracts: [{
+      name: $contractName
+      abi: {
+        json: $abi
+      }
+      compilation: {
+        id: $compilationId
+      }
+      sourceContract: {
+        index: 0
+      }
+      constructor: {
         createBytecode: {
           id: $bytecodeId
         }
-        compilation: {
-          id: $compilationId
+      }
+    }]
+  }) {
+    contracts {
+      id
+      name
+      sourceContract {
+        name
+        source {
+          contents
         }
-        contract: {
-          id: $contractId
+        ast {
+          json
         }
-      }]
-    }) {
-      contractConstructors {
-        id
+      }
+      constructor {
         createBytecode {
           bytes
-          sourceMap
-        }
-        compilation {
-          compiler {
-            name
-          }
-          contracts {
-            name
-          }
-          sources {
-            sourcePath
-          }
-        }
-        contract {
-          name
-          source {
-            sourcePath
-          }
         }
       }
     }
+  }
 }`
 
-it("adds contract constructors", async () => {
-  const workspace = new Workspace();
+describe("Contract", () => {
+  const client = new WorkspaceClient();
 
   const variables = {
-    bytecodeId: soliditySha3(Migrations.bytecode), 
-    compilationId: "0xfd63faebf4e7515b2f48fdf22066869e09df9b2e91f4e0cf88d7d14ec496b515",
-    contractId: soliditySha3(Migrations.contractName, soliditySha3(Migrations.source, Migrations.sourcePath))
+    contractName: Migrations.contractName,
+    compilationId: compilationId,
+    id: contractId,
+    compilerName: Migrations.compiler.name,
+    compilerVersion: Migrations.compiler.version,
+    sourceId: sourceId,
+    bytecodeId: bytecodeId, 
+    abi: JSON.stringify(Migrations.abi)
   }
 
-  const constructorId = soliditySha3(variables.bytecodeId);
-
-  // add contract constructors
-  {
-    const result = await graphql(
-      schema, AddContractConstructors, null, { workspace }, variables
-    );
-
-    const { data } = result;
-    expect(data).toHaveProperty("contractConstructorsAdd");
-
-    const { contractConstructorsAdd } = data;
-    expect(contractConstructorsAdd).toHaveProperty("contractConstructors");
-
-    const { contractConstructors } = contractConstructorsAdd;
-
-    expect(contractConstructors).toHaveLength(1);
-
-    const { id } = contractConstructors[0];
-    expect(id).toEqual(constructorId);
+  beforeAll(async () => {
```

Comment:
Let's just make this `before` rather than `beforeAll`, so that it runs before each `it`, just in case.

Reply:
hey @gnidan I don't see a `before()` in the Jest documentation -- just `beforeEach` and `beforeAll`. Maybe you mean I should change this to `beforeEach`, which runs before each of the tests, while `beforeAll` just runs once?

Reply:
Ah, yep, `beforeEach`!